app/lib/actions/foreman_wreckingball/host/remediate_vmware_operatingsystem.rb

```diff
@@ -34,12 +34,16 @@ def run
             :architecture => host.architecture.name,
             :osfamily => host.operatingsystem.family,
             :name => host.operatingsystem.name,
-            :major => host.operatingsystem.major.to_i
+            :major => host.operatingsystem.major.to_i,
+            :release => host.facts['os::release::full']
```

Comment:
Can we get this without accessing the facts? Does `host.operatingsystem.release` contain the same content?

Reply:
In short, no.

```
> host.operatingsystem.release                                                                                                                                                                                                                                     
=> "10.0.14393"
> host.operatingsystem.release_name
=> ""
```

The only way without directly accessing facts would probably be to have some kind of minor version database mapping version ranges to OS releases, or do matching against the end of the OS description/title.

```
> host.operatingsystem.description
=> "Windows Server 2016"
```

But as far as I know there's nothing requiring said description/title to be correct.

Reply:
Yeah, I actually do prefer facts here. What happens if `host.facts['os::release::full']` is nil? Do we need to catch this somewhere?

Reply:
Yeah, probably best to skip selectors with `nil` value in `VsphereOsIdentifiers.where`, I'll throw that check in.